server/server.go

```diff
@@ -133,6 +135,35 @@ func (s *Server) isUnixSocket() bool {
 	return s.cfg.Socket != ""
 }
 
+func (s *Server) forwardUnixSocketToTCP(socket string, addr string) {
+	if sock, err := net.Listen("unix", socket); err == nil {
+		log.Infof("Server redirecting [%s] to [%s]", socket, addr)
+		for {
+			if uconn, err := sock.Accept(); err == nil {
+				log.Infof("server socket forwarding from [%s] to [%s]", socket, addr)
+				go s.handleForwardedConnection(uconn, addr)
+			} else {
+				log.Warningf("server failed to forward from [%s] to [%s], err: %s", socket, addr, err)
+			}
+		}
+	} else {
+		log.Fatalf("err: %s", err) // in use?
+	}
+}
+
+func (s *Server) handleForwardedConnection(uconn net.Conn, addr string) {
+	if tconn, err := net.Dial("tcp", addr); err == nil {
+		if _, err := io.Copy(tconn, uconn); err != nil {
+			log.Warningf("socket forward copy failed: %s", err)
+		}
+	} else {
+		log.Warningf("socket forward failed: could not connect to [%s], err: %s", addr, err)
+	}
+	if err := uconn.Close(); err != nil {
```

Comment:
a close is usually done in a defer statement.

Reply:
If I use defer, I have to use an anonymous function to catch the error though.  it looks longer?

Reply:
We don't care much about the error returned by the Close method.
So we use
```
defer terror.Call(uconn.Close)
```